extensions/interactions/FractionInput/FractionInput.js

```diff
@@ -0,0 +1,56 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for Fraction Input rules.
+ */
+oppia.factory('fractionInputRulesService', [
+  'FractionObjectFactory',
+  function(FractionObjectFactory) {
+    var toFloat = function(fractionDict) {
+      return FractionObjectFactory.fromDict(fractionDict).toFloat();
+    };
+
+    return {
+      IsEquivalentTo: function(answer, inputs) {
+        return toFloat(answer) === toFloat(inputs.f);
+      },
+      IsEquivalentToAndInSimplestForm: function(answer, inputs) {
+        var simplestForm =
+          FractionObjectFactory.fromDict(inputs.f).convertToSimplestForm();
+        return toFloat(answer) === toFloat(inputs.f) &&
+          JSON.stringify(answer) === JSON.stringify(simplestForm);
```

Comment:
Mm, this is a bit hacky. Maybe angular.equals()? Or you could define an .isIdentical(otherFraction) method on the Fraction class.

Reply:
Yep, I changed it to angular.equals(). Done.